zio-schema/shared/src/main/scala/zio/schema/Schema.scala

```diff
@@ -46,7 +46,10 @@ sealed trait Schema[A] {
   /**
    * Returns a new schema that modifies the type produced by this schema to be optional.
    */
-  def optional: Schema[Option[A]] = Schema.Optional(self)
+  def optional(implicit ev: CanBeOptional[A]): Schema[Option[A]] = {
+    val _ = ev
```

Comment:
There's got to be a better way to do this so the compiler doesn't complain about unused arguments but I don't know how....